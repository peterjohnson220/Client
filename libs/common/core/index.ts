export * from './pipes';
export * from './helpers';
