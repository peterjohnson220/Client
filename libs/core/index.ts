export * from './helpers';
export * from './pipes';
export * from './common.module';
