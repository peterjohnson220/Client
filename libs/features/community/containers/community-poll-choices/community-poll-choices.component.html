<div class="form-group row" [formGroup]="item">
      
      <div *ngIf="isAdmin" class="col-sm-1">
            <label [attr.for]="'response'+index" class="form-control no-border">{{index+1}}.</label>
      </div>

      <div class="col-sm-9">
            <input type="text" class="form-control" [placeholder]=getPlaceholder [attr.id]="'response'+index" formControlName="response" [maxLength]="45">    
      </div>

      <div *ngIf="enableEditingResponseOptions && index > 1">
            <div class="col-sm-1">
                  <button type="button" class="btn btn-secondary border-0" (click)="removed.emit(index)"><i class="fas fa-times text-muted"></i></button>
            </div>
      </div>
</div>
