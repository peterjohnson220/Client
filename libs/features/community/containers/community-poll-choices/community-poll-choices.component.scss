@import 'colors';

.no-border {
    border: 0;
    box-shadow: none;
}

.input-containter
{
    position: relative;
    height: 30px;
}

.input-body
{
    position: absolute;
}

.remove-btn
{
    position: absolute;
    height: 100%;
    right: 0;
    padding: 0;
    border: none;
    background-color: transparent;
}

.add-btn
{
    position: absolute;
    border: none;
    padding: 0;
    background-color: transparent;
    color: $community-tab-active-color;
}
