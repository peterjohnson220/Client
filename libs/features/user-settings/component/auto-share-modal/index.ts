export * from './auto-share-modal.component';
