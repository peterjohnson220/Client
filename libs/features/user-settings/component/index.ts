export * from './auto-share-modal';
export * from './auto-shared-users';
export * from './shareable-user';
