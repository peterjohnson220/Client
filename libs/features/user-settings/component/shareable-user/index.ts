export * from './shareable-user.component';
