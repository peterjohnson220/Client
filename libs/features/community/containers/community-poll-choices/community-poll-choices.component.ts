import { Component, EventEmitter, Input, Output } from '@angular/core';
import { FormGroup, FormControl, Validators } from '@angular/forms';

@Component({
  selector: 'pf-community-poll-choices',
  templateUrl: './community-poll-choices.component.html',
  styleUrls: ['./community-poll-choices.component.scss']
})
export class CommunityPollChoicesComponent {

  static enableEditingResponseOptionsStatic = true;

  get enableEditingResponseOptions() { return CommunityPollChoicesComponent.enableEditingResponseOptionsStatic; }
  get getPlaceholder() { return this.index <= 1 ? `Choice ${this.index + 1}` : `Choice ${this.index + 1} (Optional)`; }

  @Input() public index: number;

  @Input() public item: FormGroup;

  @Input() public isAdmin: false;

  @Output() public removed: EventEmitter<number> = new EventEmitter<number>();

  static buildItem(val: string, enableEditingResponseOptions = true) {

    CommunityPollChoicesComponent.enableEditingResponseOptionsStatic = enableEditingResponseOptions;
    const responseOptionControl = new FormControl(val, [Validators.required]);

    if (!enableEditingResponseOptions) {
      responseOptionControl.disable();
    }
    return new FormGroup({
      response: responseOptionControl
    });

  }
}
