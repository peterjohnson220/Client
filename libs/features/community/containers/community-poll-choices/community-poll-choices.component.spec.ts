import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CommunityPollChoicesComponent } from './community-poll-choices.component';

describe('CommunityPollChoicesComponent', () => {
  let fixture: ComponentFixture<CommunityPollChoicesComponent>;
  let instance: CommunityPollChoicesComponent;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        ReactiveFormsModule
      ],
      declarations: [
        CommunityPollChoicesComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(CommunityPollChoicesComponent);
    instance = fixture.componentInstance;
  });

  it('should show community start poll', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should not allow to update poll choices if !enableEditingResponseOptions', () => {
    CommunityPollChoicesComponent.enableEditingResponseOptionsStatic = false;
    expect(fixture).toMatchSnapshot();
  });

  it('should not allow to update poll choices if enableEditingResponseOptions === true', () => {
    CommunityPollChoicesComponent.enableEditingResponseOptionsStatic = true;
    expect(fixture).toMatchSnapshot();
  });
});
