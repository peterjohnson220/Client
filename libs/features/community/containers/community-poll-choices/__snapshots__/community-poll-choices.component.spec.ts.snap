// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityPollChoicesComponent should not allow to update poll choices if !enableEditingResponseOptions 1`] = `
<pf-community-poll-choices
  removed={[Function EventEmitter]}
>
  <div
    class="form-group row"
  >
    
    <div
      class="col-sm-9"
    >
      <input
        class="form-control"
        formcontrolname="response"
        type="text"
      />
    </div>
    
  </div>
</pf-community-poll-choices>
`;

exports[`CommunityPollChoicesComponent should not allow to update poll choices if enableEditingResponseOptions === true 1`] = `
<pf-community-poll-choices
  removed={[Function EventEmitter]}
>
  <div
    class="form-group row"
  >
    
    <div
      class="col-sm-9"
    >
      <input
        class="form-control"
        formcontrolname="response"
        type="text"
      />
    </div>
    
  </div>
</pf-community-poll-choices>
`;

exports[`CommunityPollChoicesComponent should show community start poll 1`] = `
<pf-community-poll-choices
  removed={[Function EventEmitter]}
>
  <div
    class="form-group row"
  >
    
    <div
      class="col-sm-9"
    >
      <input
        class="form-control"
        formcontrolname="response"
        type="text"
      />
    </div>
    
  </div>
</pf-community-poll-choices>
`;
